File: examples/persist-session/rest/csharp/PersistSession.cs
// Creates a long-lived Browserless session and demonstrates persisting state across connections.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

var token = "YOUR_API_TOKEN_HERE";
var client = new HttpClient();

var content = new StringContent(
    JsonSerializer.Serialize(new { ttl = 300000, stealth = true, headless = false }),
    Encoding.UTF8,
    "application/json"
);

var response = await client.PostAsync(
    $"https://production-sfo.browserless.io/session?token={token}",
    content
);
var body = await response.Content.ReadAsStringAsync();
var session = JsonSerializer.Deserialize<JsonElement>(body);

var connectUrl = session.GetProperty("connect").GetString();
var stopUrl = session.GetProperty("stop").GetString();
Console.WriteLine($"Session created: {session.GetProperty("id")}");
Console.WriteLine($"Connect URL: {connectUrl}");

// Stop the session when done
await client.DeleteAsync($"{stopUrl}&force=true");
Console.WriteLine("Session stopped.");

File: examples/persist-session/rest/java/PersistSession.java
// Creates a long-lived Browserless session and demonstrates persisting state across connections.
//
// Run: javac PersistSession.java && java PersistSession

import java.net.URI;
import java.net.http.*;

public class PersistSession {
    public static void main(String[] args) throws Exception {
        String token = "YOUR_API_TOKEN_HERE";
        String endpoint = "https://production-sfo.browserless.io/session?token=" + token;

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(
                "{\"ttl\": 300000, \"stealth\": true, \"headless\": false}"
            ))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        String body = response.body();
        System.out.println("Session created: " + body);

        // Parse connect and stop URLs from response
        int connectStart = body.indexOf("\"connect\":\"") + 11;
        int connectEnd = body.indexOf("\"", connectStart);
        String connectUrl = body.substring(connectStart, connectEnd);

        int stopStart = body.indexOf("\"stop\":\"") + 8;
        int stopEnd = body.indexOf("\"", stopStart);
        String stopUrl = body.substring(stopStart, stopEnd);

        System.out.println("Connect URL: " + connectUrl);

        // Stop the session when done
        HttpRequest deleteRequest = HttpRequest.newBuilder()
            .uri(URI.create(stopUrl + "&force=true"))
            .DELETE()
            .build();

        client.send(deleteRequest, HttpResponse.BodyHandlers.ofString());
        System.out.println("Session stopped.");
    }
}
